src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -473,6 +486,9 @@ protected void deleteUser(User user) {
         studentScoreRepository.deleteAllByUserId(user.getId());
         learningGoalProgressRepository.deleteAllByUserId(user.getId());
         exerciseHintActivationRepository.deleteAllByUser(user);
+        postRepository.deleteAllByAuthor(user);
```

Comment:
This should fix #6331 then, too, right?

Reply:
I think these changes don't affect course deletion. 🤔  To solve #6331, the `delete(Course course)` function in `CourseService` should be changed instead.

Reply:
Oh, you are right. I misremembered my own issue. :facepalm:  I thought that was also about user deletion.